csharp/src/sdk/Geometry/Polygon.cs

```diff
@@ -322,41 +224,190 @@ internal static bool Touches(Polygon p1, Polygon p2)
         /// <returns></returns>
-        internal static IEnumerable<Polygon> Ops(Polygon p1, Polygon p2, PolygonOps op)
+        public IEnumerable<Polygon> Union(Polygon polygon)
         {
-            var p1Path = ToClipperPath(p1);
-            var p2Path = ToClipperPath(p2);
+            var thisPath = PolygonExtensions.ToClipperPath(this);
+            var polyPath = PolygonExtensions.ToClipperPath(polygon);
             Clipper clipper = new Clipper();
-            clipper.AddPath(p1Path, PolyType.ptSubject, true);
-            clipper.AddPath(p2Path, PolyType.ptClip, true);
+            clipper.AddPath(thisPath, PolyType.ptSubject, true);
+            clipper.AddPath(polyPath, PolyType.ptClip, true);
             var solution = new List<List<IntPoint>>();
-            switch (op)
+            clipper.Execute(ClipType.ctUnion, solution);
+            var polygons = new List<Polygon>();
+            foreach (List<IntPoint> path in solution)
             {
-                case PolygonOps.Difference:
-                    clipper.Execute(ClipType.ctDifference, solution);
-                    break;
-                case PolygonOps.Intersection:
-                    clipper.Execute(ClipType.ctIntersection, solution);
-                    break;
-                case PolygonOps.Union:
-                    clipper.Execute(ClipType.ctUnion, solution);
-                    break;
-                case PolygonOps.XOR:
-                    clipper.Execute(ClipType.ctUnion, solution);
-                    break;
+                polygons.Add(PolygonExtensions.ToPolygon(path));
             }
+            return polygons;
+        }
+
+        /// <summary>
+        /// Returns Polygons representing the symmetric difference between this Polygon and the supplied Polygon.
+        /// </summary>
+        /// <param name="polygon">The intersecting polygon.</param>
+        /// <returns></returns>
+        public IEnumerable<Polygon> XOR(Polygon polygon)
+        {
+            var thisPath = PolygonExtensions.ToClipperPath(this);
+            var polyPath = PolygonExtensions.ToClipperPath(polygon);
+            Clipper clipper = new Clipper();
+            clipper.AddPath(thisPath, PolyType.ptSubject, true);
+            clipper.AddPath(polyPath, PolyType.ptClip, true);
+            var solution = new List<List<IntPoint>>();
+            clipper.Execute(ClipType.ctUnion, solution);
             var polygons = new List<Polygon>();
             foreach (List<IntPoint> path in solution)
             {
-                polygons.Add(ToPolygon(path));
+                polygons.Add(PolygonExtensions.ToPolygon(path));
             }
             return polygons;
         }
+
+        /// <summary>
+        /// Offset this polyline by the specified amount.
+        /// </summary>
+        /// <param name="offset">The amount to offset.</param>
+        /// <returns>A new polyline offset by offset.</returns>
```

Comment:
I never updated this to Polygon, but see what this does? `A new polygon..` It tells us we're getting a new Polygon *not* just updating the state of the target.